### DISCUSSION TITLE
Rai Pal - Portable

### DISCUSSION THREAD

**Discussion by sirjmann92** [Raicuparta/rai-pal#186] [Ideas]
Is it possible to run this application as a portable? I'd much rather have all files in one directory. It would make it easier to manage game profiles and mods and keep everything contained in a single directory. As it is now, the game is installed in one place, mods and data files are stored in another (hidden AppData directory), and then we have to decide where to store/keep game profiles for UEVR (if we want to use them), which is likely a third directory/downloads folder for many users.

Also, within the "RaiPal" root installation directory there is a "resources" subdirectory, and within that is a "uevr" directory. Inside the "uevr" directory is a txt file "place_uevr_files_here.txt". I'll be using this folder for profiles but it's a bit confusing because the actual UEVR mod data is stored elsewhere (AppData)...

If I've missed some documentation with instruction on portable installation, forgive me. I did look before posting.

**Comment by Raicuparta** [comment]
I somehow missed this discussion entirely.

To clarify for anyone reading this in the future, that UEVR directory with the `place_uevr_files_here.txt` file is a legacy thing from back when you had to download UEVR manually and add it to Rai Pal. It shouldn't be there in new Rai Pal versions anymore, but that leftover folder will stay there unless you delete it yourself.

One reason for the lack of a portable mode is due to Rai Pal being made with [Tauri](https://tauri.app/), which itself doesn't officially support a portable mode.

Another reason is making it harder to auto-update. I'd have to implement my own system for it, which I've actually done before, but still would be some extra work plus could be potentially more fragile. I already have issues with people using old versions of Rai Pal even when it nags you about a new version being available, and then reporting bugs I've already fixed.

All that said, I'm not ruling out the possibility of having a portable version. Or at the very least letting users configure the folder where all mods etc get placed. Once that's done, you can sorta just move the Rai Pal exe around and it will work on any system that has the required dependencies installed.
